client/src/partials/budget/analysis/analysis_budget.js

```diff
@@ -0,0 +1,420 @@
+      .then(loadFiscalYears);
+    }
+
+    function addBudgetData() {
+      // Insert the budget numbers into the account data
+      // TODO: The following procedural hacks can be simplified by better SQL queries...
+      // First compute the totals for any accounts with budgets
+      var totalBudget = 0.0;
+      var totalBalance = 0.0;
+      var totals = {};
+      $scope.budgets.data.forEach(function (bud) {
+        if (bud.account_id in totals) {
+          totals[bud.account_id] += bud.budget;
+        }
+        else {
+          totals[bud.account_id] = bud.budget;
+        }
+        totalBudget += bud.budget;
+      });
+
+      var accounts_data_id = [];
+      $scope.accounts.data.forEach(function (acct) {
+        accounts_data_id.push(acct.id);
+      });
+
+      // Previous Budgets processing
+      // Total budget per fiscal year
+      var totalsFY = {},
+          totalBudgetFY = {};
+      session.selectedPreviousFY.forEach(function (fy) {
+        totalsFY[fy.id] = {};
+        totalBudgetFY[fy.id] = 0.0;
+        $scope.fiscalYearBudget[fy.id].forEach(function (bud) {
+          if (bud.account_id in totalsFY[fy.id]) {
+            totalsFY[fy.id][bud.account_id] += bud.budget;
+          }
+          else {
+            totalsFY[fy.id][bud.account_id] = bud.budget;
+          }
+          if (accounts_data_id.indexOf(bud.account_id) !== -1) {
+            totalBudgetFY[fy.id] += bud.budget;
+          }
+        });
+      });
+
+
+      // Insert the budget totals into the account data
+      $scope.accounts.data.forEach(function (acct) {
+        // Current budget
+        if (acct.id in totals) {
+          acct.budget = precision.round(totals[acct.id], 2);
+        } else {
+          if (acct.type === 'title') {
+            acct.budget = null;
+          } else {
+            acct.budget = 0; // No budget means 0 budget!
+          }
+        }
+
+        // Previous Budgets
+        acct.previousBudget = {};
+        session.selectedPreviousFY.forEach(function (fy) {
+          if (acct.id in totalsFY[fy.id]) {
+            acct.previousBudget[fy.id] = precision.round(totalsFY[fy.id][acct.id], 2);
+          } else {
+            if (acct.type === 'title') {
+              acct.previousBudget[fy.id] = null;
+            } else {
+              acct.previousBudget[fy.id] = 0; // No budget means 0 budget!
+            }
+          }
+        });
+
+        // Increment total balance
+        if (!isNaN(acct.balance)) {
+          totalBalance += acct.balance;
+        }
+      });
+
+      $scope.total.budget = precision.round(totalBudget, 2);
+      $scope.total.balance = precision.round(totalBalance, 2);
+      // Previous Budget Data
+      $scope.totalPreviousBudget = {};
+      session.selectedPreviousFY.forEach(function (fy) {
+        $scope.totalPreviousBudget[fy.id] = Math.round(totalBudgetFY[fy.id], 2);
+      });
+    }
+
+    function parseAccountDepth(accountData, accountModel) {
```

Comment:
This function does two things:
 1. Parses the account depth and assigns account.depth to equal the depth in the tree
 2. Totals the surplus/deficit based on the account classe

These are two vary different pieces of logic.  Do you think they might belong in two different functions?